packages/core/src/tracing/opentelemetry-instrumentations/wrap.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+// @ts-nocheck
+
+'use strict';
+
+const semver = require('semver');
+const { AsyncHooksContextManager } = require('@opentelemetry/context-async-hooks');
+const api = require('@opentelemetry/api');
+const { BasicTracerProvider } = require('@opentelemetry/sdk-trace-base');
+const constants = require('../constants');
+const instrumentations = {
+  '@opentelemetry/instrumentation-fs': { name: 'fs' },
+  '@opentelemetry/instrumentation-restify': { name: 'restify' },
+  '@opentelemetry/instrumentation-socket.io': { name: 'socket.io' }
+};
+
+// NOTE: using a logger might create a recursive execution
+//       logger.debug -> creates fs call -> calls transformToInstanaSpan -> calls logger.debug
+//       use uninstrumented logger, but useless for production
+module.exports.init = (_config, cls) => {
+  // CASE: otel offically does not support Node < 14
+  // https://github.com/open-telemetry/opentelemetry-js/tree/main#supported-runtimes
+  if (semver.lt(process.versions.node, '14.0.0')) {
+    return;
+  }
+
+  Object.keys(instrumentations).forEach(k => {
+    const value = instrumentations[k];
+    const instrumentation = require(`./${value.name}`);
+    instrumentation.init(cls);
+    value.module = instrumentation;
+  });
+
+  const transformToInstanaSpan = otelSpan => {
+    if (!otelSpan || !otelSpan.instrumentationLibrary) {
+      return;
+    }
+
+    const targetInstrumentionName = otelSpan.instrumentationLibrary.name;
+    let kind = constants.EXIT;
+
+    if (instrumentations[targetInstrumentionName] && instrumentations[targetInstrumentionName].module) {
+      const targetInstrumentationModule = instrumentations[targetInstrumentionName].module;
+
+      if (targetInstrumentationModule.getKind) {
+        kind = targetInstrumentationModule.getKind(otelSpan);
+      }
+
+      if (targetInstrumentationModule.transform) {
+        otelSpan = targetInstrumentationModule.transform(otelSpan);
+      }
+    }
+
+    if ((kind === constants.EXIT && cls.skipExitTracing()) || (kind === constants.ENTRY && cls.tracingSuppressed())) {
+      return;
+    }
```

Comment:
@basti1302 Would love to get your input and help here.

As discussed in the tracer meeting. It was asked about entry and exit span conditions.
I have added this line to ensure we respect the rules.

I am wondering if we need anything else here, but I cannot think of any other check 🤔 
For example the http server instrumentation creates an entry span and does not care wether there is already an existing entry span or not.

FYI we create entry spans only for Socket.io ON events.

Reply:
Discussed offline. See commits.